File: src/main/java/org/dxworks/utils/java/rest/client/providers/BasicAuthenticationProvider.java
package org.dxworks.utils.java.rest.client.providers;

import com.google.api.client.http.BasicAuthentication;
import com.google.api.client.http.HttpRequest;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class BasicAuthenticationProvider implements AuthenticationProvider {
    private String username;
    private String password;

    @Override
    public void initialize(HttpRequest httpRequest) {
        httpRequest.setInterceptor(new BasicAuthentication(username, password));
    }
}

File: src/main/kotlin/org/dxworks/utils/java/rest/client/providers/BasicAuthenticationProvider.kt
package org.dxworks.utils.java.rest.client.providers

import com.google.api.client.http.HttpRequest

open class BasicAuthenticationProvider(
    protected val username: String,
    protected val password: String
) : AuthenticationProvider {

    override fun initialize(httpRequest: HttpRequest) {
        httpRequest.headers.setBasicAuthentication(username, password)
    }
}
